### PR TITLE
Add deprecate description

### LINKED ISSUE #95: change frontend to request kube-apiserver directly to CRUD resources and deprecate all APIs for CRUD resources
Currently, when the simulator API is requested to create a resource, it requests the change to kube-apiserver via client-go. 
So, our backend API can be considered as a kind of BFF (backend for frontend), acting as an intermediary between kube-apiserver and the frontend.

However, for the API to create/read/update/delete resources, the simulator API is just passing the request to the kube-apiserver without doing any additional processing. In other words, simple requests such as creating, deleting, and editing resources can be thrown directly to kube-apiserver by the frontend without going through the backend.

This change will make it easier to extend in the future. (like supporting new resource type)

/kind feature

### PR DESCRIPTION
**Pull request by momom-i**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Once we change this simulator's frontend to request resource change directly to kube-apiserver, all APIs on server side can be deprecated.

#### Which issue(s) this PR fixes:

Fixes #95

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -60,6 +60,10 @@ empty
 | 202   | |
 | 500 | something went wrong (see logs of the simulator server) |
 
+---
+## **deprecated APIs**
```
Comment (sanposhiho) [inline docs/api.md:64]:
```suggestion
## deprecated APIs
```

File: docs/api.md [hunk 2]
```diff
@@ -60,6 +60,10 @@ empty
...
+
+The following APIs are deprecated. They will be removed in the near future.
 ## Apply Node
```
Comment (sanposhiho) [inline docs/api.md:67]: Could you change all deprecated APIs titles like this?

```suggestion
### Apply Node
```

**Review by sanposhiho — COMMENTED** [review]
only two nits

**Comment by sanposhiho** [comment]
I think you can resolve lint error with `nolint` like this.
`//nolint: staticcheck // we need to use deprecated APIs for now.`

Add this `nolint` on here:
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/server/server.go#L24-L26

File: docs/api.md (hunk 1 above)
Comment (momom-i) [reply docs/api.md]: fixed: [4d227ee](https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/127/commits/4d227ee0e035741bd10e0ad809bd2ee610ced078)

File: docs/api.md (hunk 2 above)
Comment (momom-i) [reply docs/api.md]: fixed: [4d227ee](https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/127/commits/4d227ee0e035741bd10e0ad809bd2ee610ced078)

**Comment by momom-i** [comment]
Thank you for your review and I'm sorry for ignoring error💦

**Review by sanposhiho — APPROVED** [review]
Thanks!
/lgtm
/approve

Let's wait #115 
/hold

**Pull request merged**